### PR TITLE
feat: add check run actions

### PR DESCRIPTION
**Pull request by anjakammer**
Actions add custom events to a check run.

### CODE REVIEW

**Comment by anjakammer** [comment]
Sorry, these are my first lines of GO I ever made. I don't know how it comes, that the indentation is wrong here.
I already tested this functionality here: https://github.com/anjakammer/node-http/pull/66/checks?check_run_id=62403716

Be kind and do not run re-run some checks. :)

File: pkg/check/check_run.go [hunk 1]
```diff
@@ -107,3 +110,10 @@ type Image struct {
 	Alt      string `json:"alt"`
 	Caption  string `json:"caption,omitempty"`
 }
+
+// Action is an custom event definition
+type Action struct {
+	Label 			string `json:"label"`
+	Identifier 	string `json:"identifier"`
+	Description string `json:"description"`
```
Comment (radu-matei) [inline pkg/check/check_run.go]: This is where the linter is unhappy - lines 116 and 117, the type and tag (`string `json:"label"``) should be aligned.

Comment (anjakammer) [reply pkg/check/check_run.go]: I needed to run `gofmt -s`, the code looks fine in my IDE. :thinking:

**Comment by radu-matei** [comment]
Also, could you please make sure you sign the CLA?

**Comment by anjakammer** [comment]
@radu-matei 
I use actions to provide a 'rollback' or 'delete deployment' button for the Deployment stage of my pipeline. So after a new pipeline stages finishes, I execute this Job:
```Javascript
class SendSignal extends Job {
  constructor ({ stage, logs, conclusion, actions }) {
    super(`result-of-${stage}`.toLowerCase(), checkRunImage)
    this.storage.enabled = false
    this.imageForcePull = true
    this.useSource = false
    this.env = {
      CHECK_PAYLOAD: payload,
      CHECK_NAME: stage,
      CHECK_TITLE: 'Description',
      CHECK_CONCLUSION: conclusion,
      CHECK_SUMMARY: `${stage} ${conclusion}`,
      CHECK_TEXT: logs
    }
    if (typeof actions !== 'undefined') {
      this.env.CHECK_ACTIONS = JSON.stringify(actions)
    }
  }
}
```
A action for a rollback would look like this:
```Javascript
      'actions': [{
        'label': 'rollback',
        'identifier': 'rollback_deployment',
        'description': 'rollback this deployment'
      }]
```
The important parts of the custom action webhook payload would look like this:
```Javascript
{
  "action": "requested_action",
  "check_run": {
    "id": 62403475
    },
// [...]
  "requested_action": {
    "identifier": "rollback_deployment"
  }
// [...]
} 
```

But the gateway clearly needs to deal with this action type. For now, one can only create this actions.

**Comment by anjakammer** [comment]
Actually, the gateway is fine with this webhook, since it is `X-GitHub-Event: check_run`
Works like a charm :)

**Pull request merged**